tools/tpm2_nvdefine.c

```diff
@@ -95,7 +102,25 @@ static int nv_space_define(tpm_nvdefine_ctx *ctx) {
 
     // Now set the attributes.
     public_info.t.nvPublic.attributes.val = ctx->nvAttribute;
-    public_info.t.nvPublic.authPolicy.t.size = 0;
+    if (ctx->enforce_read_policy) {
+        public_info.t.nvPublic.attributes.TPMA_NV_POLICYREAD = 1;
+    }
+    if (ctx->enforce_write_policy) {
+        public_info.t.nvPublic.attributes.TPMA_NV_POLICYWRITE = 1;
+    }
+    if (ctx->enforce_delete_policy) {
+        public_info.t.nvPublic.attributes.TPMA_NV_POLICY_DELETE = 1;
+    }
+
+    if (!ctx->policy_file_flag) {
+        public_info.t.nvPublic.authPolicy.t.size = 0;
+    } else {
+        public_info.t.nvPublic.authPolicy.t.size  = sizeof(public_info.t.nvPublic.authPolicy) - 2;
```

Comment:
on line 60, do TPM2B_EMPTY INIT
```
 TPM2B_NV_PUBLIC public_info = TPM2B_EMPTY_INIT;
```
Then here, just do:
```
if (ctx->policy_file_flag) {
   public_info.t.nvPublic.authPolicy.t.size = BUFFER_SIZE(TPM2B_NV_PUBLIC, <buffername>);
    files_load_bytes_from_file();
   ...
}
```

This gets rid of the if clause of your branch.